addons/addon-user-id-upgrade/packages/user-id-upgrade-post-deployment/lib/data/tables.js

```diff
@@ -0,0 +1,38 @@
+const { listValue, common } = require('../helpers/transforms');
+const passwordsTransforms = require('./password-transforms');
+const userApiKeysTransforms = require('./user-api-keys-transforms');
+const keyPairsTransforms = require('./key-pairs-transforms');
+const studyPermissionsTransforms = require('./study-permissions-transforms');
+const workflowDraftsTransforms = require('./workflow-drafts-transforms');
+
+const tables = () => [
+  { name: 'DeploymentStore', keys: ['type', 'id'], transforms: common },
+  { name: 'Passwords', keys: ['username'], transforms: [...passwordsTransforms, ...common] },
```

Comment:
In this specific case, the key is username.  This is related to how the exiting design of the Passwords table is.  Maybe @dpandya has more context around this table.

Reply:
Are we saying that we don't want to rename the table's column, since 'username' is used to access it? I think now would be a good time to rename the table since we are creating new tables (right?)

Reply:
This is passwords table and comes into play only during AuthN in case of "internal" or "local" users. This table is used when validating user's password when the user logs in using internal AuthN provider (i.e., using username/password). The `uid` does not come into play at this point. Once the user is authenticated correctly (i.e., once the `username`/`password` combination is validated) we then lookup the user in the `User` table using the `username` and `ns` using the GSI named `Principal` and find the logged in user's `uid`. That GSI allows for looking up users the old way i.e., using the `username` and `ns` combination. This GSI exists mainly for the AuthN flow. We do not have `uid` during AuthN as we do not know who user is (that's the point of the AuthN :)). Once the AuthN is complete we have `uid` and then `uid` is used as single identifier in the rest of the code. We should not rename this table and this column. The password table should continue to have `username` as the partition key.

Reply:
Perfect thanks!